### PR TITLE
Initial framework

### PR DESCRIPTION
**Pull request by vm**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    fn hello_world(_: &mut Request) -> IronResult<Response> {
+        Ok(Response::with(status::Ok))
+    }
+
+    fn add_link(req: &mut Request) -> IronResult<Response> {
+        let mut payload = String::new();
+        request.body.read_to_string(&mut payload).unwrap();
+        println!("request: {}", payload);
+        let link = serde_json::from_str(&payload).unwrap();
```
Comment (jamesmcnamara) [inline src/main.rs]: What happens with link here?

Comment (vm) [inline src/main.rs]: nothing yet, i wanna add a db add here. trying to figure out how that works.

File: src/main.rs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    router.post("/add", add_link);
+    Iron::new(router).http("localhost:9000").unwrap();
+
+    fn hello_world(_: &mut Request) -> IronResult<Response> {
+        Ok(Response::with(status::Ok))
+    }
+
+    fn add_link(req: &mut Request) -> IronResult<Response> {
+        let mut payload = String::new();
+        request.body.read_to_string(&mut payload).unwrap();
```
Comment (vm) [inline src/main.rs]: is there a way to do this without mut payload? maybe something other than read_to_string?

File: src/main.rs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+struct Link {
+    address: String,
+    sender: String,
+    message: String
+}
+
+fn main() {
+    let mut router = Router::new();
+    router.get("/", hello_world);
+    router.post("/add", add_link);
```
Comment (vm) [inline src/main.rs]: is this the same as
```rs
let router = Router::new()
    .get("/", hello_world)
    .post("/add", add_link);
Iron::new(router).http("localhost:9000").unwrap();
```

File: src/main.rs (hunk 2 above)
Comment (jamesmcnamara) [inline src/main.rs]: Check out the [Read](https://doc.rust-lang.org/std/io/trait.Read.html) trait.

File: src/main.rs (hunk 3 above)
Comment (jamesmcnamara) [inline src/main.rs]: Yes. You can see in the documentation that the methods mutate the router to bind the callback behavior also return `&mut Router`. This idiom is designed for chaining.

On a side note, I'm not sure how I feel about that pattern, from a framework designer's POV. Python is super against it, but C++ is super for it. Rust is oddly in the middle.

Comment (vm) [inline src/main.rs]: Maybe they just want good method chaining, aka readability?

**Comment by vm** [comment]
final changes for the initial stuff. accept if you like.

**Pull request merged**